### PR TITLE
Bugfixed, published in this page:

### PR DESCRIPTION
**Pull request by GlobalManager**
https://github.com/OCA/server-tools/issues/1179

And Chilean translation added.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
You are mixing several things in the same patch. Please change only the needed for solving the issue, and you can change translations in Transifex: https://odoo-community.org/page/translate

**Comment by GlobalManager** [comment]
Some errors were encountered, related to the version of npm in the server, for example:

/home/travis/.npm/_logs/2018-05-28T01_33_46_147Z-debug.log

**Comment by GlobalManager** [comment]
@gurneyalex 
@pedrobaeza 

Watch this. Some errors were encountered, related to the version of npm in the server, for example:

/home/travis/.npm/_logs/2018-05-28T01_33_46_147Z-debug.log

**Comment by pedrobaeza** [comment]
The error seems to be due to too many requests.

Any way, I see that you are changing too much things and removing features of the module, so please keep the minimal diff for fixing the problem. If there are several things to fix, only one per PR.

**Comment by GlobalManager** [comment]
All fixes are uploaded.

**Comment by GlobalManager** [comment]
You can merge whenever you can. Thanks.

**Comment by GlobalManager** [comment]
Can you check this?. @pedrobaeza

Thanks

File: auditlog/models/http_request.py
Comment (sebalix) [inline auditlog/models/http_request.py]: Is this empty line required? It introduces a useless change which can create trouble when backporting patches between branches.

File: auditlog/models/http_request.py
Comment (sebalix) [inline auditlog/models/http_request.py]: Same remark, is this empty line required?

File: auditlog/models/http_request.py
Comment (sebalix) [inline auditlog/models/http_request.py]: This piece of code was added maybe in 8.0 or 9.0 to avoid some error, are you sure it is not required anymore? You got some problem with it?

File: auditlog/models/http_session.py
Comment (sebalix) [inline auditlog/models/http_session.py]: Is this empty line required?

File: auditlog/models/http_session.py
Comment (sebalix) [inline auditlog/models/http_session.py]: Is this empty line required?

File: auditlog/models/rule.py
Comment (sebalix) [inline auditlog/models/rule.py]: `_logger` does not seem to be used in this file.

File: auditlog/models/rule.py
Comment (sebalix) [inline auditlog/models/rule.py]: This code is to be compatible with old API which can always be present in a Odoo 10.0 project (modules not standard nor in OCA), that's why the signature of the method was `*args, **kwargs` to inspect the content of `args`.
If your call to `read(...)` is made with the new API, the rule creation is already satisfied below line 264 ("New API", https://github.com/OCA/server-tools/pull/1259/files#diff-e2cdd2f4b9da04f93936a15fa53fef79R269).

File: auditlog/models/rule.py
Comment (sebalix) [inline auditlog/models/rule.py]: If the only change is the condition `if res_name and res_name != "False"`, limit the diff by rewriting it to use the `continue` statement as we are in a `for` loop.
But this change also need some clarification, have you an example of how you encountered a "False" string value?

**Review by sebalix — CHANGES_REQUESTED** [review]
For sure some bugs need to be fixed in `auditlog`. Some remarks, but in general we need some explanation to understand the different fixes proposed here.

File: auditlog/models/rule.py
Comment (GlobalManager) [reply auditlog/models/rule.py]: There is, always, a case when creates a register of the action performed, but creates two of them each time, one with (example): res.partned(1,), and the other the same res.partned(1,), but the first the attributes in False, like the name, although it's the object with the correct ID: res.partned(1,).

File: auditlog/models/rule.py
Comment (GlobalManager) [reply auditlog/models/rule.py]: I can let the signature as *args, but the signature *args, **kwargs, was what was generating the error, fixed by this patch. The actual signature for read in Odoo 10.0, is:

def read(self, fields=None, load='_classic_read'): #Two parameters

I can let the signature like:
def read(self, *args):

So the next sentence doesn't make sense in this case: cr, uid, ids = args[0], args[1], args[2]

I guess read in Odoo 10.0, isn't compatible with old api anymore. The module is working.

File: auditlog/models/rule.py
Comment (sebalix) [reply auditlog/models/rule.py]: Can you write
```python
if not res_name or res_name == "False":
   continue
```
to avoid an indentation level? It makes the diff very big (and harder to backport too).

**Review by sebalix — CHANGES_REQUESTED** [review]

**Comment by GlobalManager** [comment]
@sebalix We did some changes and fix the feature uploaded.

The reads, writes, creates and unlinks were going to be registered even if the read, create, write, unlink, weren't checked. We fixed this issue.

**Comment by GlobalManager** [comment]
To reproduce the issue with the last code of auditlog: Set the rule for res.users and subscribe it, in the rules view, without marking any checkbox. Then perform one read of res.users, that can be made reloading the Odoo's webpage. The read will be registered, even if the read checkbox wasn't checked.

**Comment by StefanRijnhart** [comment]
@GlobalManager can you create a separate PR for that last bugfix? BTW. the idea is that if the rule is not enabled for an operation, the model is not patched for that method: https://github.com/OCA/server-tools/blob/10.0/auditlog/models/rule.py#L131 Can you fix the problem there instead?

**Comment by GlobalManager** [comment]
But there are 4 checkbox to be checked, if the checkbox aren't checked, the rules shouldn't be executed. Is the normal flow of a understandable module.

**Comment by GlobalManager** [comment]
Create patching method is working in the current patch to be merged: https://github.com/OCA/server-tools/blob/10.0/auditlog/models/rule.py#L131

**Comment by StefanRijnhart** [comment]
Thanks for isolating that one! Let's continue the discussion on the 11.0 port that you already created here: https://github.com/OCA/server-tools/pull/1264

**Comment by GlobalManager** [comment]
Once it is merged, I will apply the others changes, you told me to delete.

**Comment by GlobalManager** [comment]
Because, the rest of the changes depend of the changes in this PR.

File: auditlog/models/rule.py
Comment (sebalix) [reply auditlog/models/rule.py]: Okay, I'm testing a piece of code with an `read` overload in old API, and it seems it's not compatible anymore with the new API in Odoo 10.0.
So OK to drop the support for old API. But regarding the fix, the one proposed here is wrong. The `_make_read` method should be written like this:
```python

        @api.multi
        def read(self, fields=None, load='_classic_read'):
            result = read.origin(self, fields, load)
            read_values = dict((d['id'], d) for d in result)
            # If the call came from auditlog itself, skip logging:
            # avoid logs on `read` produced by auditlog during internal
            # processing: read data of relevant records, 'ir.model',
            # 'ir.model.fields'... (no interest in logging such operations)
            if self.env.context.get('auditlog_disabled'):
                return result
            self = self.with_context(auditlog_disabled=True)
            rule_model = self.env['auditlog.rule']
            rule_model.sudo().create_logs(
                self.env.uid, self._name, self.ids,
                'read', read_values, None, {'log_type': log_type})
            return result

        return read
```

File: auditlog/models/rule.py
Comment (sebalix) [inline auditlog/models/rule.py]: Why changing this? the environment has already the `auditlog_disabled` set to True (two lines above).

File: auditlog/models/rule.py
Comment (sebalix) [inline auditlog/models/rule.py]: Do not change these lines, this is not necessary.

File: auditlog/models/rule.py
Comment (sebalix) [inline auditlog/models/rule.py]: Do not change these lines, this is not necessary.

File: auditlog/models/rule.py
Comment (sebalix) [inline auditlog/models/rule.py]: Do not change these lines, this is not necessary.

**Review by sebalix — CHANGES_REQUESTED** [review]
Comment done bellow.
Please avoid to change lines of code not related to the fix, it makes the diff bigger (hard to read, and possibly hard to backport in other branches).

**Comment by GlobalManager** [comment]
The test_pylint applied during the tests, is giving the error, but there is not have to be with auditlog.

**Comment by GlobalManager** [comment]
What we have to

**Comment by GlobalManager** [comment]
If the pylint failed, what can we do?.

**Comment by sebalix** [comment]
The lint failed but it's not related to auditlog, so do nothing about it. If a lint was related to auditlog, just fix it :)

**Comment by GlobalManager** [comment]
That's nice.

File: auditlog/models/rule.py
Comment (sebalix) [inline auditlog/models/rule.py:246]: `result`/`result2` trick isn't necessary anymore as the `read` method is only compatible with the new API. Check the piece of code I sent in the other comment.

**Review by sebalix — CHANGES_REQUESTED** [review]

**Pull request closed without merging**
